CoreSchemaView/src/au/gov/asd/tac/constellation/views/schemaview/providers/TransactionTypeNodeProvider.java

```diff
@@ -130,15 +143,98 @@ public void newActiveGraph(final Graph graph) {
             } else {
                 schemaLabel.setText("No schema available");
             }
+            populateTree();
+        });
+    }
 
-            final TreeItem<SchemaTransactionType> root = createNode(null);
-            treeView.setRoot(root);
+    private VBox addFilter() {
+        filterText.setPromptText("Filter transaction types");
+        final ToggleGroup tg = new ToggleGroup();
+        startsWithRb.setToggleGroup(tg);
+        startsWithRb.setPadding(new Insets(0, 0, 0, 5));
+        startsWithRb.setSelected(true);
+        final RadioButton containsRb = new RadioButton("Contains");
+        containsRb.setToggleGroup(tg);
+        containsRb.setPadding(new Insets(0, 0, 0, 5));
+
+        tg.selectedToggleProperty().addListener((ov, oldValue, newValue) -> {
+            populateTree();
         });
+
+        filterText.textProperty().addListener((ov, oldValue, newValue) -> {
+            populateTree();
+        });
+
+        final HBox headerBox = new HBox(new Label("Filter: "), filterText, startsWithRb, containsRb);
+        headerBox.setAlignment(Pos.CENTER_LEFT);
+        headerBox.setPadding(new Insets(5));
+
+        final VBox box = new VBox(schemaLabel, headerBox, treeView);
+        VBox.setVgrow(treeView, Priority.ALWAYS);
+        return box;
+    }
+
+    private void populateTree() {
+        final TreeItem<SchemaTransactionType> root = createNode(null);
+        treeView.setRoot(root);
+    }
+
+    private boolean isFilterMatchCurrentNode(SchemaTransactionType treeItem) {
+        boolean found = isFilterMatchText(treeItem.getName());
+        if (!found) {
+            found = isFilterMatchAnyProperty(treeItem);
+        }
+        return found;
+    }
+
+    private boolean isFilterMatchAnyProperty(SchemaTransactionType treeItem) {
+        boolean found = false;
+        final String name = treeItem.getName();
```

Comment:
This block could be made more compact using something like
```
return isFilterMatchText(treeItem.getName()) 
    || isFilterMatchText(treeItem.getDescription())
    ...
    || treeItem.getProperties.stream().anyMatch(property -> 
        property != null && isFilterMatchText(property.toString())
    );
```
...although I do wonder how much readability you'd lose.

Reply:
I'll add my 2 cents here and say that provided you added each condition on a new line, I think what @cygnus-x-1 has suggested is more readable than a large list of if statements. The numerous amount of if statements will also increase the cognitive complexity higher than I reckon it needs to be

Reply:
@Auriga2 did you see this message?